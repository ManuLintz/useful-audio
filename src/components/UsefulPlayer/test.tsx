import { render, screen } from '@testing-library/react'

import Avatar from '.'

describe('<Avatar />', () => {
  const props = {
    src: 'https://gravatar.com/4405735f6f3129e0286d9d43e7b460d0',
    alt: 'Avatar'
  }

  it('should render the medium Avatar as default', () => {
    const { container } = render(<Avatar {...props} />)

    expect(screen.getByRole('img', { name: /Avatar/i })).toBeInTheDocument()

    expect(container.firstChild).toHaveClass(
      'inline-block w-12 h-12 rounded-full'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the upload file invitation', () => {
    const { container } = render(<Avatar />)

    expect(screen.getByTestId('upload-file')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
