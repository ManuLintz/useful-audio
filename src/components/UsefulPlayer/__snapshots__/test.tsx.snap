// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Avatar /> > should render the medium Avatar as default 1`] = `
<img
  alt="Avatar"
  class="inline-block rounded-full w-12 h-12"
  src="https://gravatar.com/4405735f6f3129e0286d9d43e7b460d0"
/>
`;
